server/models/AssessmentVariablesModel/assessmentVariablesModel.test.js

```diff
@@ -0,0 +1,202 @@
+      var3: 'str',
+    }),
+  ]
+  beforeAll(async () => {
+    appDb = await global.MONGO_INSTANCE.db('assessment_vars')
+  })
+  beforeEach(async () => {
+    await appDb.createCollection(collections.metadata)
+    await appDb.createCollection(collections.assessmentDayData)
+    await appDb.createCollection(collections.assessments)
+    await appDb.createCollection(collections.assessmentVariables)
+  })
+
+  afterEach(async () => {
+    const existingCollections = await appDb.listCollections().toArray()
+    const collectionNames = existingCollections.map((c) => c.name)
+    if (collectionNames.includes(collections.metadata)) {
+      await appDb.collection(collections.metadata).drop()
+    }
+    if (collectionNames.includes(collections.assessmentDayData)) {
+      await appDb.collection(collections.assessmentDayData).drop()
+    }
+    if (collectionNames.includes(collections.assessments)) {
+      await appDb.collection(collections.assessments).drop()
+    }
+    if (collectionNames.includes(collections.assessmentVariables)) {
+      await appDb.collection(collections.assessmentVariables).drop()
+    }
+  })
+  afterAll(async () => {
+    await appDb.dropDatabase()
+  })
+  it('Updates the assessment variables when there are new variables', async () => {
+    const newAssessment = 'update-vars-assessment'
+    const assessmentAttributes = AssessmentModel.withDefaults({
+      name: newAssessment,
+    })
+    const assessmentDoc = await AssessmentModel.upsert(
+      appDb,
+      { name: newAssessment },
+      assessmentAttributes
+    )
+    const intialVariables = initialAssessmentVariables.map(({ name }) => ({
+      name,
+      assessment_id: assessmentDoc._id,
+    }))
+    const newAssessmentVariables = [
+      createAsessmentVariable({
+        name: 'var1',
+      }),
+      createAsessmentVariable({
+        name: 'var2',
+      }),
+      createAsessmentVariable({
+        name: 'var3',
+      }),
+      createAsessmentVariable({
+        name: 'var4',
+      }),
+      createAsessmentVariable({
+        name: 'var6',
+      }),
+      createAsessmentVariable({
+        name: 'var7',
+      }),
+      createAsessmentVariable({
+        name: 'var5',
+      }),
+      createAsessmentVariable({
+        name: 'new-var1',
+      }),
+      createAsessmentVariable({
+        name: 'newvar2',
+      }),
+    ]
+
+    const data = createNewAssessmentDayData({
+      metadata: createAssessmentDayDataMetadata({
+        ...assessmentDayDataControllerMetadata,
+        assessment: newAssessment,
+      }),
+      participant_assessments: initialParticipantDayData,
+      assessment_variables: newAssessmentVariables,
+    })
+    const request = createRequest({
+      body: data,
+      app: { locals: { appDb } },
+    })
+    const response = createResponse()
+
+    await appDb
+      .collection(collections.assessmentVariables)
+      .insertMany(intialVariables)
+
+    await AssessmentDayDataController.create(request, response)
```

Comment:
We shouldn't be calling the controller actions from the model test. These test examples should be scoped to only the model function logic/functionality.

Reply:
yeah, but @mikestone14 we chatted about this and I would like to ensure that these updates are happening appropriately, if these tests don't belong in the controller and if they belong in the model then how can we ensure that this logic is working.

Reply:
The controller tests can assert that the model is being called with the correct parameters. The model tests can assert that given various params the data is changed as expected. Does that make sense?

Reply:
> I agree I can spy on the method and verify that it is called with the right arguments and that will always be true, because we are mapping through all of the assessment variables in the payload

This makes sense to have as a test because it _should_ always be true but if we change the parameters or the logic it will break and that's a good thing.

> I think that to assert that there is no duplication we want to compare the results of variables that were previously inserted from a past import, and now we have a new import with payload of old and new assessment variables

This seems to make sense as a model tests where we can have test examples that have no existing data, some existing data (some of which have changed), and existing data that does not need to change. This way we can make sure the app handles all these cases as we expect.

Reply:
The controller should also have a test that asserts on the JSON response of the API call, so that might also cover part of what you mentioned above regarding what we'd likely see on the page.

Reply:
phew, here it is... 
spy tests [55385f4](https://github.com/AMP-SCZ/dpdash/pull/732/commits/55385f44f0326ff3200eb6258e408e23d9199dd8)

model tests [c512de7](https://github.com/AMP-SCZ/dpdash/pull/732/commits/c512de781b56d695f539f619fc0d0f1f1eb0ae74)